rotkehlchen/db/filtering.py

```diff
@@ -84,23 +84,33 @@ def prepare(self) -> Tuple[List[str], List[Any]]:
 
 
 @dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=False)
-class DBETHTransactionAddressFilter(DBFilter):
+class DBETHTransactionJoinsFilter(DBFilter):
     """Find transactions involving any of the addresses. Including in an internal.
 
     This uses the mappings we create in the DB at transaction addition to signify
     relevant addresses for a transaction.
     """
-    addresses: List[ChecksumEthAddress]
+    addresses: Optional[List[ChecksumEthAddress]]
+    join_events: bool = False
 
     def prepare(self) -> Tuple[List[str], List[Any]]:
-        questionmarks = '?' * len(self.addresses)
-        filters = [
-            f'AS A LEFT OUTER JOIN ethtx_address_mappings AS B WHERE '
-            f'A.tx_hash=b.tx_hash AND B.address IN ({",".join(questionmarks)})',
-        ]
-        bindings = self.addresses
+        query_filter, bindings = '', []
+        if self.join_events:
+            query_filter += (
+                ' INNER JOIN '
+                '(SELECT event_identifier, counterparty, asset FROM history_events)'
+                'ON hex(ethereum_transactions.tx_hash)=substr(event_identifier, 3)'
+            )
+        if self.addresses:
+            questionmarks = '?' * len(self.addresses)
+            query_filter += str(
+                f'INNER JOIN ethtx_address_mappings ON '
```

Comment:
This seems broken. If both `self.join_events` and `self.addresses` have truthy values you end up with a broken string: `'...(event_identifier, 3)INNER JOIN ...'`

Please fix this, and think about this a bit. This is the reason we have filters in a list and we join them later. Like this the code is super brittle and easy to misunderstand and break.

What's more this means that all the combinations are definitely not tested in this PR. So please create unit tests of all filter combinations of this one.

Reply:
Yes, I understand, will improve tests and fix the query.

Filters can have only one possible `join_clause` property so we with @yabirgb found this as the best way to do multiple joins.